### ISSUE TITLE
Bug: Throwing TypeError (Version 1.2.6)

### ISSUE THREAD

**Issue by RonanC** [Evilweed/protractor-beautiful-reporter#115]
**Working Version:** `1.2.5`  
**Broken Version:** `1.2.6`  
**NPM Repo:** https://www.npmjs.com/package/protractor-beautiful-reporter  

This morning I went to run Protractor and got the below error.
I noticed that the version of this module was updated last night.
This bug must have been introduced.

### env
**OS:** `Windows 7 64bit`
**Chrome version:** `Version 70.0.3538.77 (Official Build) (64-bit)`

### Error:
```yaml
[15:30:48] E/launcher - Error: TypeError: Cannot set property 'searchSettings' of undefined
    at new ScreenshotReporter (C:\Users\RonanC\development\myApp\node_modules\protractor-beautiful-reporter\index.js:4672:47)
    at onPrepare (C:\Users\RonanC\development\myApp\e2e\protractor.conf.js:82:38)
    at q_1.Promise (C:\Users\RonanC\development\myApp\node_modules\protractor\built\util.js:46:49)
    at Function.promise (C:\Users\RonanC\development\myApp\node_modules\q\q.js:682:9)
    at Object.runFilenameOrFn_ (C:\Users\RonanC\development\myApp\node_modules\protractor\built\util.js:38:16)
    at plugins_.onPrepare.then (C:\Users\RonanC\development\myApp\node_modules\protractor\built\runner.js:98:27)
    at _fulfilled (C:\Users\RonanC\development\myApp\node_modules\q\q.js:834:54)
    at C:\Users\RonanC\development\myApp\node_modules\q\q.js:863:30
    at Promise.promise.promiseDispatch (C:\Users\RonanC\development\myApp\node_modules\q\q.js:796:13)
    at C:\Users\RonanC\development\myApp\node_modules\q\q.js:556:49
[15:30:48] E/launcher - Process exited with error code 100
An unexpected error occurred: undefined
```

**Comment by RonanC** [comment]
Yup, that's what I'm doing for now.

**Comment by miller45** [comment]
I research the issue immediately..,

**Comment by miller45** [comment]
Damn actually I wanted to avoid this inconvience but now it is unavoidable...another solution (using the 1.2.6 version) is to migrate your searchsettings (in protractor conf) to a new syntax now you have so say:
```
 , clientDefaults:{
       searchSettings:{
           allselected: false,
           passed: false,
           failed: true,
           pending: true,
           withLog: true
       }
   }
```
instead of (old version which crashes 1.2.6)
```
 searchSettings:{
           allselected: false,
           passed: false,
           failed: true,
           pending: true,
           withLog: true
       }
```
Sorry for that...

**Comment by miller45** [comment]
I you have searchSettings **and** columnSettings the new format looks likes this:
```
onPrepare: function () {
        // Add a screenshot reporter:
        jasmine.getEnv().addReporter(new HtmlReporter({
            baseDirectory: 'reports-tmp',
            clientDefaults: {
                searchSettings: {
                    allselected: false,
                    passed: false,
                    failed: true,
                    pending: true,
                    withLog: true
                },
                columnSettings:{
                    displayTime:true,
                    displayBrowser:false,
                    displaySessionId:false,
                    inlineScreenshots:false
                }
            }
        }).getJasmine2Reporter());
    },
```

**Comment by miller45** [comment]
@RonanC Thank you for the full callstack...helped me to isolate the issue very fast...

**Comment by miller45** [comment]
If you have no searchSettings/clientSettings at all i might also crash...in that case just provide and empty object:
```
onPrepare: function () {
        // Add a screenshot reporter:
        jasmine.getEnv().addReporter(new HtmlReporter({
            baseDirectory: 'reports-tmp',
            clientDefaults: {     
            }
        }).getJasmine2Reporter());
    },
```

**Comment by miller45** [comment]
For people with no (prior) configuration of `searchSettings` or `columnSettings`, the `clientDefaults:{}` "fix" is just for the time being, and is not required in the next version(s).

**Comment by miller45** [comment]
People with "old" searchSettings or columnSettings are forced to the new syntax (with clientDefaults) because of the bug.

**Comment by miller45** [comment]
@RonanC Concerning next release: I have send an email to the repositorys owner for a new npm release. Usually it takes a few days (maximum) for him to "react".

**Comment by Evilweed** [comment]
@miller45 I'll release new version when I'll get to my computer. (probably in next 2-3 hours)

**Comment by miller45** [comment]
@RonanC @matukaz is fixed now in version 1.2.7

**Comment by RonanC** [comment]
Thanks +1
